### ISSUE TITLE
Add implicit cast from floatN/halfN to UnityEngine.Color and back

### ISSUE THREAD

**Issue by henningboat** [Unity-Technologies/Unity.Mathematics#105]
It would be nice to be able to implicitly cast from a color to a vector and back. 
Currently I'm using an extension method like this, but I it would be nicer to just cast from one to the other.

> public static class Extensions
	{
		public static float3 ToFloat3(this Color c)
		{
			return new float3(c.r, c.b, c.b);
		}
	}

**Comment by aybe** [comment]
Indeed, that'd be great if `Color` could implicitly be casted to `float4` as well.
